src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -411,7 +412,7 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     completionWork onFailure {
       case e =>
         logger.error(s"Preemption work failed for call ${callKey.tag}! " + e.getMessage, e)
-        scheduleTransition(WorkflowFailed)
+        self ! CheckForWorkflowComplete
```

Comment:
What about just checking for completion when you receive a `PersistenceSucceeded` / `Failed` message ? I think this is a time when you're supposed to have a real representation of the world.
It just seems to me that there is a lot of those `CheckForWorkflowComplete` flying around. For example in something like this
```scala
  persistStatusThenAck(callKey, ExecutionStatus.Failed, sender(), message, callOutputs = None, returnCode = returnCode)
  val updatedData = data.addPersisting(callKey, ExecutionStatus.Failed)
  self ! CheckForWorkflowComplete
  stay() using updatedData
```
persistStatusThenAck will send a `PersistenceSucceeded` (or Failure) message which will send already a `CheckForWorkflowComplete` message, so I don't see the point of re-sending one in this code.